virtualmachine/vsphere/util.go

```diff
@@ -693,6 +784,22 @@ var uploadTemplate = func(vm *VM, dcMo *mo.Datacenter, selectedDatastore string)
 			return fmt.Errorf("snapshot task returned an error: %s", err)
 		}
 	} else {
+		if vm.Destination.DestinationType == DestinationTypeHost {
+			if len(vm.Disks) > 0 {
+				if err = reconfigureVM(vm, vmMo); err != nil {
+					return err
+				}
+			}
+			// power on
+			if err = start(vm); err != nil {
+				return err
+			}
+			fmt.Println("VM is powered on, waiting for ip")
+			if err = waitForIP(vm, vmMo); err != nil {
+				return err
+			}
+			return nil
```

Comment:
In case of esxi host, it tries to create the template vm if it does not exists and fails as marking vm as template on esxi host is not supported. So we need to provision the vm if the template doesn't exist. If template exists it will clone the template to provision vm.